lib/Server.js

```diff
@@ -376,25 +377,26 @@ class Server {
       });
     } else {
       // route content request
-      this.app.get(
-        '*',
+      this.app.use(
+        contentBasePublicPath,
         express.static(contentBase, this.options.staticOptions)
       );
     }
   }
 
   setupServeIndexFeature() {
     const contentBase = this.options.contentBase;
+    const contentBasePublicPath = this.options.contentBasePublicPath;
 
     if (Array.isArray(contentBase)) {
       contentBase.forEach((item) => {
-        this.app.get('*', serveIndex(item));
+        this.app.use(contentBasePublicPath, serveIndex(item));
       });
     } else if (
       typeof contentBase !== 'number' &&
       !isAbsoluteUrl(String(contentBase))
     ) {
-      this.app.get('*', serveIndex(contentBase));
+      this.app.use(contentBasePublicPath, serveIndex(contentBase));
```

Comment:
Here interesting situations:
-  Before we use `get` so handle only `get` requests, now we use `use` so we handle all requests
- `*` means handle all request, `/` handle only `index page`, imagine you use url `/my/webpack-dev-server`, now contentBase doesn't work

Reply:
/cc @imranismail

Reply:
@evilebottnawi Should tests catch this use case?

Reply:
As I tested
When using `this.app.use` with `/` it means all paths starting with `/`  
When using `this.app.get` with `/` it means only index page  
When using `this.app.get` with `*` it means all paths

Reply:
Also `serveIndex` and `serveStatic` will handle only `GET` requests, others will be passed to `next` middleware

Reply:
hm, i think you are right, but yes let's add tests :+1:

Reply:
@iamandrewluca let's tests this too

Reply:
@evilebottnawi can you take a look at `test/server/contentBasePublicPath-option.test.js` I added?  
It seems should cover this case.

Reply:
Ok, I'll add some tests for other methods than `GET`